calculator.c

```diff
@@ -0,0 +1,148 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+#include <math.h>
+
+#define MAX 100
+
+// Stack structure
+typedef struct {
+    int items[MAX];
+    int top;
+} Stack;
+
+void push(Stack *s, int value) {
+    if (s->top < MAX - 1) {
+        s->items[++(s->top)] = value;
+    } 
+    else {

```

Comment:
The stack push function is meant for adding element to stack. Error handling should be done outside this function. Make functions do one task not multiple tasks. This applies throughout the code.

Reply:
👍